app/src/main/java/com/nervousfish/nervousfish/cryptography/KeyGenerator.java

```diff
@@ -0,0 +1,62 @@
+package com.nervousfish.nervousfish.cryptography;
+
+import java.security.KeyFactory;
+import java.security.KeyPair;
+import java.security.KeyPairGenerator;
+import java.security.NoSuchAlgorithmException;
+import java.security.spec.InvalidKeySpecException;
+import java.security.spec.RSAPrivateKeySpec;
+import java.security.spec.RSAPublicKeySpec;
+
+public class KeyGenerator {
+
+    /**
+     * Generates a random KeyPair with the RSA algorithm.
+     * @return a randomly generated KeyPair
+     */
+    public static KeyPair generateRandomKeyPair() {
+        try {
+            KeyPairGenerator k = KeyPairGenerator.getInstance("RSA");
+
+            k.initialize(2048);
+            return k.generateKeyPair();
+        } catch (NoSuchAlgorithmException e) {
+            e.printStackTrace();
+        }
+        return null;
```

Comment:
I think it's better to throw a new Exception("explanation of what went wrong") than to return null, because now the program just continues and we may not even notice that something went wrong